drivers/scheduler/scheduler.go

```diff
@@ -44,6 +44,9 @@ type Driver interface {
 	// IsNodeReady checks if node is in ready state. Returns nil if ready.
 	IsNodeReady(n node.Node) error
 
+	// GetNodes returns all nodes known the scheduler
+	GetNodes() ([]node.Node, error)
```

Comment:
we don't need the GetNodes() method in the scheduler as the node package already has it.
Check drivers/node/node_registry.go
The scheduler adds all nodes in the node registry at the start and all the consumers are using GetNodes, GetWorkerNodes from this.